### PR TITLE
Ship env vars as one-time secret

### PR DESCRIPTION
**Pull request by csweichel**
## Description
Today user environment variables become environment variables on the workspace pod. This is a clean and straight forward way to implement env vars, but it suffers from a drawback: values are stored in clear-text on the pod. If one isn't careful about logging, those values could be spread into the logs, and are visible to everyone with access to the cluster.

This PR starts shipping env vars as one-time secret, much like we ship the Git hoster token. This way, environment variable values are not visible to anyone, except within the workspace.

## How to test
1. Set up user and project level environment variables.
2. Start a workspace and ensure those env vars are present
3. Run `kubectl get pod -o yaml ws-...` and ensure the user env vars are not present.

## Release Notes

```release-note
Improved in-transit security of user environment variables
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **filiptronicek**, **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @filiptronicek @geropl` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by geropl** [comment]
/lgtm

Code looks good, tested and works as advertised.

/hold
But IMO we should keep the served times in-sync with the ones we use for other OTS data ([example](https://github.com/gitpod-io/gitpod/blob/9dc26a3deb43def5934f663b412677138dcd97e2/components/server/src/workspace/workspace-starter.ts#L941)): `30` minutes. Otherwise I fear we'd introduce strange (and new) failure modes.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 1f24a3d385fdb8e9898472e4319b46676e7c6d32</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7923#issuecomment-1029135102" title="LGTM">geropl</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **filiptronicek** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @filiptronicek` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]
- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by csweichel** [comment]
@geropl fair point - will correct this

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by csweichel** [comment]
@geropl changed the env var OTS timeout - needs another lgtm

**Comment by geropl** [comment]
I hoped for a constant. :smile: But 
/lgtm
anways

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 60750202a474f0d6177f69539a5f16bc7546bdb0</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7923#issuecomment-1029781754" title="LGTM">geropl</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **filiptronicek** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @filiptronicek` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]
- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7923#issuecomment-1029781918" title="Approved">geropl</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **filiptronicek** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @filiptronicek` in a comment when ready.

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/7923#issuecomment-1029781918" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]
- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by csweichel** [comment]
> I hoped for a constant.

That would have been smarter. Lesson learned: first coffee, then PR

**Comment by akosyakov** [comment]
I cannot really test it, it fails when i try to configure Gitpod app in order to create a new project:
Screenshot 2022-02-04 at 15 16 29

**Comment by csweichel** [comment]
looking at the code you'll see that we treat the `allEnvvars` list just as before, i.e. if project-level env vars worked before it's very likely they'll work still.

**Comment by akosyakov** [comment]
/lgtm

I rely on @geropl testing. 😄  Scanned for compatibilities between server and supervisor looks like can be deployed in any order.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7923#issuecomment-1030043546" title="LGTM">akosyakov</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7923#issuecomment-1029781918" title="Approved">geropl</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/7923#issuecomment-1029781918" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]
- ~~[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)~~ [akosyakov]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/workspace/workspace-starter.ts [hunk 1]
```diff
@@ -603,12 +603,31 @@ export class WorkspaceStarter {
...
+        // TODO(cw): for the time being we're still pushing the env vars as we did before.
+        //           Once everything is running with the latest supervisor, we can stop doing that.
+        allEnvVars.forEach(e => {
             const ev = new EnvironmentVariable();
-            ev.setName(uv.name);
-            ev.setValue(uv.value);
-            return ev;
-        });
+            ev.setName(e.name);
+            ev.setValue(e.name);
```
Comment (jankeromnes) [inline components/server/src/workspace/workspace-starter.ts:628]: Oops:

Screenshot 2022-02-08 at 10 15 40

Comment (jankeromnes) [reply components/server/src/workspace/workspace-starter.ts:628]: Also sometimes the value seems to be `name + ' ' + value`? 🤔

Screenshot 2022-02-08 at 10 17 08

[the remark this answers is not in the document]
Comment (geropl) [reply components/server/src/workspace/workspace-starter.ts:628]: How did this work then during testing, at all? :thinking:

[the remark this answers is not in the document]
Comment (jankeromnes) [reply components/server/src/workspace/workspace-starter.ts:628]: @geropl I think you [tested](https://github.com/gitpod-io/gitpod/pull/7923#issuecomment-1029135102) before the bug [was introduced](https://github.com/gitpod-io/gitpod/pull/7923#commits-pushed-198c07f).

**Pull request merged**